pfcpiface/conn.go

```diff
@@ -179,12 +179,26 @@ func (pConn *PFCPConn) setLocalNodeID(id string) {
 
 // Serve serves forever a single PFCP peer.
 func (pConn *PFCPConn) Serve() {
-	go func() {
+	connClosed := make(chan struct{}, 1)
+	go func(connClosed chan struct{}) {
 		recvBuf := make([]byte, 65507) // Maximum UDP payload size
 
 		for {
+			err := pConn.SetReadDeadline(time.Now().Add(pConn.upf.readTimeout))
+			if err != nil {
+				log.Errorf("failed to set read timeout: %v", err)
+			}
+
 			n, err := pConn.Read(recvBuf)
 			if err != nil {
+				if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+					log.Infof("Read timeout received for connection: %v<->%v",
```

Comment:
Thinking about who will have to interpret these logs:
```suggestion
					log.Infof("Read timeout for connection %v<->%v, is the SMF still alive?",
```